langchain4j-core/src/main/java/dev/langchain4j/data/document/Document.java

```diff
@@ -158,7 +105,46 @@ public static Document document(String text) {
      * @param metadata the metadata of the document.
      * @return a new Document.
      */
-    public static Document document(String text, Metadata metadata) {
+    static Document document(String text, Metadata metadata) {
         return from(text, metadata);
     }
+
+    /**
+     * DefaultDocument is an implementation of the Document interface.
+     * It represents an unstructured piece of text with associated metadata.
+     */
+    record DefaultDocument(
```

Comment:
Do we want to keep it nested in the `Document` interface? I would move it to it's own file